Framework/PythonInterface/plugins/algorithms/MedianBinWidth.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.api import AlgorithmFactory, HistogramValidator,\
+    MatrixWorkspaceProperty, PythonAlgorithm
+from mantid.kernel import Direction
+import numpy
+import roundinghelper
+
+
+class MedianBinWidth(PythonAlgorithm):
+
+    _PROP_BIN_WIDTH = 'BinWidth'
+    _PROP_INPUT_WS = 'InputWorkspace'
+
+    def category(self):
+        '''
+        Return algorithm's category.
+        '''
+        return 'Utility\\Calculation'
+
+    def name(self):
+        '''
+        Return algorithm's name.
+        '''
+        return 'MedianBinWidth'
+
+    def summary(self):
+        '''
+        Return algorithm's summary.
+        '''
+        return ("Calculates the average of workspace's histograms'"
+                " median bin widths.")
+
+    def version(self):
+        '''
+        Return algorithm's version.
+        '''
+        return 1
+
+    def PyInit(self):
+        '''
+        Declares algorithm's properties.
+        '''
+        self.declareProperty(
+            MatrixWorkspaceProperty(name=self._PROP_INPUT_WS,
+                                    defaultValue='',
+                                    validator=HistogramValidator(),
+                                    direction=Direction.Input),
+            doc='The workspace containing the input data')
+        roundinghelper.declare_rounding_property(self)
+        self.declareProperty(self._PROP_BIN_WIDTH,
+                             defaultValue=0.0,
+                             direction=Direction.Output,
+                             doc='The averaged median bin width')
+
+    def PyExec(self):
+        '''
+        Averages the median bin widths of the input workspace.
+        '''
+        inputWs = self.getProperty(self._PROP_INPUT_WS).value
+        roundingMode = self.getProperty(
+            roundinghelper.PROP_NAME_ROUNDING_MODE).value
+        n = inputWs.getNumberHistograms()
+        medians = numpy.empty(n)
+        for wsIndex in range(n):
```

Comment:
Real testing to actually get some solid data? Okay, I tried this with data from the IN5 instrument which has 98304 spectra and 1024 bins. Mantid's `getMemorySize()` returns 2415953776 for this; Python's `meliae.scanner.get_recursive_size()` reports that `x2D` consumes 806101000 bytes. 

For timing, I get the 10 times faster speeds of `npway` for small workspaces. For the IN5 data, the speedup is less, 3.7 vs. 0.9 seconds, so 4x faster.

As a side note, using `numpy.diff()` is slightly slower than `x[1:] - x[:-1]` in the for loop.

By looking at the numbers, I think I'll implement this in the npway, as you suggested. If we run into memory problems with IN5, perhaps we need to rewrite `MedianBinWidthAtX` in C++.